scripts/workers/submission_worker.py

```diff
@@ -212,22 +213,21 @@ def extract_challenge_data(challenge, phases):
     EVALUATION_SCRIPTS[challenge.id] = challenge_module
 
 
-def load_active_challenges():
+def load_active_challenge(challenge_pk):
     '''
          * Fetches active challenges and corresponding active phases for it.
     '''
-    q_params = {'approved_by_admin': True}
+    q_params = {'approved_by_admin': True, 'pk': challenge_pk}
     q_params['start_date__lt'] = timezone.now()
     q_params['end_date__gt'] = timezone.now()
 
     # make sure that the challenge base directory exists
     create_dir_as_python_package(CHALLENGE_DATA_BASE_DIR)
 
-    active_challenges = Challenge.objects.filter(**q_params)
+    active_challenge = Challenge.objects.filter(**q_params)
```

Comment:
Rename this to `challenge`

Reply:
Also, instead of filter, we can use the get method with proper exception handling.

Reply:
sure 👍